### PR TITLE
bugfix: not report when no gpu device

### LINKED ISSUE #618: [BUG] Node.Status should not display device resource when there is no corresponding device
**What happened**:
image
这是纯 CPU 节点所显示的资源情况。

**What you expected to happen**:
理论上，节点如果没有 GPU 的话，是不应该显示 GPU 相关的 Extended resource。

/assign @ZYecho

### PR DESCRIPTION
**Pull request by ZYecho**
Signed-off-by: Yue Zhang <huaihuan.zy@alibaba-inc.com>

### Ⅰ. Describe what this PR does

fix #618 
### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/slo-controller/noderesource/noderesource.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func (r *NodeResourceReconciler) isGPUResourceNeedSync(new, old *corev1.Node) bo
 }
 
 func (r *NodeResourceReconciler) updateGPUNodeResource(node *corev1.Node, device *schedulingv1alpha1.Device) error {
-	if device == nil {
+	if device == nil || len(device.Spec.Devices) == 0 {
 		return nil
 	}
 	memoryTotal := resource.NewQuantity(0, resource.DecimalSI)
```
Comment (jasonliu747) [inline pkg/slo-controller/noderesource/noderesource.go:141]: `memoryTotal` should be shown in BinarySI format, and `coreTotal` should be DecimalSI

File: pkg/util/resource.go [hunk 2]
```diff
@@ -62,5 +62,9 @@ func IsResourceDiff(old, new corev1.ResourceList, resourceName corev1.ResourceNa
 		return true
 	}
 
+	if !oldExist {
```
Comment (jasonliu747) [inline pkg/util/resource.go:65]: are you sure? why?

File: pkg/slo-controller/noderesource/noderesource.go [hunk 3]
```diff
@@ -135,7 +135,7 @@ func (r *NodeResourceReconciler) isGPUResourceNeedSync(new, old *corev1.Node) bo
 }
 
 func (r *NodeResourceReconciler) updateGPUNodeResource(node *corev1.Node, device *schedulingv1alpha1.Device) error {
-	if device == nil {
+	if device == nil || len(device.Spec.Devices) == 0 {
```
Comment (jasonliu747) [inline pkg/slo-controller/noderesource/noderesource.go:138]: Devices 长度不等于 0⃣️ 的情况下，也不一定会包含 GPU

Comment (jasonliu747) [inline pkg/slo-controller/noderesource/noderesource.go:138]: Devices 长度不等于 0⃣️ 的情况下，也不一定会包含 GPU

**Review by jasonliu747 — CHANGES_REQUESTED** [review]

File: pkg/slo-controller/noderesource/noderesource.go (hunk 3 above)
Comment (hormes) [reply pkg/slo-controller/noderesource/noderesource.go:138]: rdma 等设备会不会出现在这里

File: pkg/util/resource.go (hunk 2 above)
Comment (hormes) [reply pkg/util/resource.go:65]: 看起来相当于是 old, new 都不存在的情形，这个写法逻辑没问题，但容易让人困惑，建议改成：
if !oldExist && !newExist {
  return nil
}

File: pkg/koordlet/statesinformer/states_device_linux.go [hunk 4]
```diff
@@ -60,6 +60,10 @@ func (s *statesInformer) reportDevice() {
 		return
 	}
 
+	if len(gpuDevices) == 0 {
```
Comment (jasonliu747) [inline pkg/koordlet/statesinformer/states_device_linux.go:63]: 这里直接return确实可以work，但合适吗😂

**Review by jasonliu747 — APPROVED** [review]
/lgtm
others PTAL

**Pull request merged**